lib/dal/src/migrations/U2600__attribute_value_dependency_graph_summary.sql

```diff
@@ -0,0 +1,1732 @@
+CREATE TABLE IF NOT EXISTS attribute_value_dependencies (
+    pk                             ident PRIMARY KEY DEFAULT ident_create_v1(),
+    id                             ident NOT NULL DEFAULT ident_create_v1(),
+    tenancy_workspace_pk           ident,
+    visibility_change_set_pk       ident NOT NULL DEFAULT ident_nil_v1(),
+    visibility_deleted_at          TIMESTAMP WITH TIME ZONE,
+    created_at                     TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT clock_timestamp(),
+    updated_at                     TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT clock_timestamp(),
+    source_attribute_value_id      ident NOT NULL,
+    destination_attribute_value_id ident NOT NULL
+);
+
+SELECT standard_model_table_constraints_v1('attribute_value_dependencies');
+INSERT INTO standard_models (table_name, table_type, history_event_label_base, history_event_message_name)
+    VALUES ('attribute_value_dependencies', 'model', 'av_deps', 'AV Deps Summary');
+
+CREATE INDEX ON attribute_value_dependencies(source_attribute_value_id);
+CREATE INDEX ON attribute_value_dependencies(destination_attribute_value_id);
+
+CREATE INDEX ON components(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON components(tenancy_workspace_pk);
+CREATE INDEX ON attribute_values(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON attribute_values(tenancy_workspace_pk);
+CREATE INDEX ON attribute_prototypes(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON attribute_prototypes(tenancy_workspace_pk);
+CREATE INDEX ON attribute_prototype_arguments(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON attribute_prototype_arguments(tenancy_workspace_pk);
+CREATE INDEX ON attribute_prototype_arguments(tail_component_id);
+CREATE INDEX ON props(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON props(tenancy_workspace_pk);
+CREATE INDEX ON prop_belongs_to_prop(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON prop_belongs_to_prop(tenancy_workspace_pk);
+CREATE INDEX ON internal_providers(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON internal_providers(tenancy_workspace_pk);
+CREATE INDEX ON external_providers(visibility_deleted_at) WHERE (visibility_deleted_at IS NOT NULL);
+CREATE INDEX ON external_providers(tenancy_workspace_pk);
+
+DROP INDEX IF EXISTS attribute_value_belongs_to_attribute_prototype_single_associati;
+CREATE UNIQUE INDEX attribute_value_belongs_to_attribute_prototype_single_associati
+    ON attribute_value_belongs_to_attribute_prototype (object_id, tenancy_workspace_pk, visibility_change_set_pk)
+    WHERE visibility_deleted_at IS NULL;
```

Comment:
The old constraint didn't have the `WHERE visibility_deleted_at IS NULL`, which meant that you couldn't `unset_belongs_to` and `set_belongs_to` for the same `object_id` within the same change set, as there would both be a soft-deleted record and a (conflicting) not-deleted record.